src/DiagnosticsProvider.php

```diff
@@ -12,125 +12,79 @@ class DiagnosticsProvider {
 
     private static $tokenKindToText;
 
+    /**
+     * @param int $kind (must be a valid token kind)
+     * @return string
+     */
+    public static function getTextForTokenKind($kind) {
+        if (!isset(self::$tokenKindToText)) {
+            self::initTokenKindToText();
+        }
+        return self::$tokenKindToText[$kind];
+    }
+
+    /**
+     * @return string[]
+     */
+    private static function initTokenKindToText() {
+        self::$tokenKindToText = \array_flip(\array_merge(
```

Comment:
Could someone explain to me what the advantage of  `array_flip(array_merge([...]))` is?

Reply:
Creating a hashmap that can be looked up in O(1)?

Reply:
oh, that's cool. Nice trick, will use that when I need it :)